cli/flox/doc/manifest.toml.md

```diff
@@ -295,6 +295,11 @@ but this behavior may change.
 It is best to write hooks defensively, assuming the user is using the
 environment from any directory on their machine.
 
+Hooks may be run non-interactively by commands other than `activate` (e.g.
+`services start`),
+so they should be written in such a way that they can be run multiple times
+non-interactively.
```

Comment:
Uggh ... that's not a great UX, and contradicts all other documentation that states the activate hook only runs once per activation.

Can we be more specific and document this as a known bug, something like:

```suggestion
⚠️ Users should be aware that at present the `hook.on-activate` script
will be invoked again with each `flox services start` invocation.
We are working to fix this issue in a future release
but in the meantime users of the service management feature
will want to ensure that `hook.on-activate` scripts are idempotent,
capable of being invoked multiple times non-interactively."
```

Reply:
I'm not aware that we're planning to fix it - if the environment changes, with our current design, don't we have to re-run them?

Reply:
Remember we said the right way to restart services following a change in the vars or activation hook is:

1. stop services
2. deactivate environment
3. reactivate environment
4. start services

So yes .. we will re-run the activate hook, but only after having performed the de-activate step; i.e. that hook won't need to be idempotent.

Reply:
I think `run multiple times non-interactively` accurately describes that. It doesn't say whether or not they have to be idempotent, just that the script may be run multiple times, which is what we will do now and plan to do in the future

Reply:
> Uggh ... that's not a great UX, and contradicts all other documentation that states the activate hook only runs once per activation.

That's what we all talked about though...

> So yes .. we will re-run the activate hook, but only after having performed the de-activate step; i.e. that hook won't need to be idempotent.

> It doesn't say whether or not they have to be idempotent, just that the script may be run multiple times, which is what we will do now and plan to do in the future

The script already gets run multiple times if you've ever activated the same environment more than once (e.g. in multiple shells), so a user may not consciously know that the script is being run more than once, but it is in fact already being run more than once and they've likely already addressed any issues that could be caused by that if they're using hooks. This is just us being explicit about the cases where it's getting run more than once.

I think a reasonable statement would be along the lines of
```
The `hook.on-activate` script is run once per activation, but as part of running some commands
we may create ephemeral activations and thus run the script multiple times.
For this reason it's considered a best practice to make your scripts idempotent.
However, only the first time the script is run (when you activate the environment) affects the
environment of your current shell.
```

Reply:
I've been stumbling into this today, i think we should be clearer as to _what_ ephemeral activations are and _when_ they are created, and at least on the implementation, _why_ we need them.
I still have some memory of discussion on that but except:

```
// TODO: there's probably a cleaner way to extract the functionality we need
// for start and restart,
// but for now just hack through the is_ephemeral bool.
```

there's not much do draw from to learn about ephemeral activations.